lib/web/apps.go

```diff
@@ -196,6 +204,22 @@ func (h *Handler) createAppSession(w http.ResponseWriter, r *http.Request, p htt
 		return nil, trace.Wrap(err)
 	}
 
+	// Ensuring proxy can handle the connection is an optional step.
+	if h.preflightAppConnection != nil && req.PreflightConnection {
+		h.log.Debugf("Ensuring proxy can handle requests requests for application %q.", identity.RouteToApp.Name)
+		err := h.preflightAppConnection(r.Context(), identity)
+		if err != nil {
+			deleteSessionErr := authClient.DeleteAppSession(r.Context(), types.DeleteAppSessionRequest{
+				SessionID: ws.GetName(),
+			})
+			if deleteSessionErr != nil {
+				h.log.WithError(err).Error("Failed to delete app session")
+			}
+
+			return nil, trace.ConnectionProblem(err, "Unable to serve application requests. Please try requesting again. If the issue persists, verify if the Application Services are connected to Teleport.")
```

Comment:
What about changing "Please try requesting again" to "Please try again"? I feel like "requesting" without an object doesn't tell the user what exactly they should try requesting again.